web_generator/lib/src/dart_main.dart

```diff
@@ -64,22 +68,51 @@ Future<void> generateJSInteropBindings({
 }
 
 Future<void> generateIDLBindings({
-  required String outputDirectory,
+  Iterable<String>? input,
+  required String output,
   required bool generateAll,
   required Version languageVersion,
 }) async {
-  const librarySubDir = 'dom';
-
-  ensureDirectoryExists('$outputDirectory/$librarySubDir');
-
-  final bindings = await generateBindings(packageRoot, librarySubDir,
-      generateAll: generateAll);
-  for (var entry in bindings.entries) {
-    final libraryPath = entry.key;
-    final library = entry.value;
-
-    final contents = _emitLibrary(library, languageVersion).toJS;
-    fs.writeFileSync('$outputDirectory/$libraryPath'.toJS, contents);
+  if (input == null) {
+    // parse dom library as normal
+    const librarySubDir = 'dom';
+
+    ensureDirectoryExists('$output/$librarySubDir');
+
+    final bindings = await generateBindings(packageRoot, librarySubDir,
+        generateAll: generateAll);
+
+    for (var entry in bindings.entries) {
+      final libraryPath = entry.key;
+      final library = entry.value;
+
+      final contents = _emitLibrary(library, languageVersion).toJS;
+      fs.writeFileSync('$output/$libraryPath'.toJS, contents);
+    }
+  } else {
+    final single = input.length == 1;
+    // parse individual files
+    ensureDirectoryExists(single ? p.dirname(output) : output);
```

Comment:
Why do these differ when there's a single input? Shouldn't we always check the dirname of the output directory?

Reply:
Ah right, this was in a previous change. It may be cleaner to always and only accept a directory for the output if we need to support the multiple input case anyways. Thoughts?